packages/sanity/src/desk/panes/document/DocumentPane.tsx

```diff
@@ -144,24 +145,26 @@ function DocumentPaneInner(props: DocumentPaneProviderProps) {
   }
 
   return (
-    <DocumentPaneProvider
-      // this needs to be here to avoid formState from being re-used across (incompatible) document types
-      // see https://github.com/sanity-io/sanity/discussions/3794 for a description of the problem
-      key={`${documentType}-${options.id}`}
-      {...providerProps}
-    >
-      {/* NOTE: this is a temporary location for this provider until we */}
-      {/* stabilize the reference input options formally in the form builder */}
-      {/* eslint-disable-next-line react/jsx-pascal-case */}
-      <ReferenceInputOptionsProvider
-        EditReferenceLinkComponent={ReferenceChildLink}
-        onEditReference={handleEditReference}
-        initialValueTemplateItems={templatePermissions}
-        activePath={activePath}
+    <CommentsEnabledProvider documentId={options.id} documentType={options.type}>
```

Comment:
Added here so that we can access the context value in `DocumentPaneProvider` and conditionally add the comment providers if comments is enabled